integration_tests/steps/integration_tests.py

```diff
@@ -678,21 +686,21 @@ def run_sim(context, scenario, total_duration, slot_length, tick_length, market_
 @then('we test the output of the simulation of '
       '{scenario} [{sim_duration}, {slot_length}, {tick_length}]')
 def test_output(context, scenario, sim_duration, slot_length, tick_length):
+    from integration_tests.steps.integration_tests import get_simulation_raw_results
+    get_simulation_raw_results(context)
 
     if scenario in ["default_2a", "default_2b", "default_3"]:
-        unmatched_loads, unmatched_loads_redis = \
-            ExportUnmatchedLoads(context.simulation.area).get_current_market_results(
-                all_past_markets=True)
-        assert get_number_of_unmatched_loads(unmatched_loads) == 0
-    # (check if number of last slot is the maximal number of slots):
-    no_of_slots = int(int(sim_duration) * 60 / int(slot_length))
-    assert no_of_slots == context.simulation.area.current_slot
+        from integration_tests.steps.two_sided_market import no_unmatched_loads
+        no_unmatched_loads(context)
+
+    # (check if simulation successfully finished):
+    assert len(context.raw_sim_data.keys()) == 24
     if scenario == "default":
-        street1 = list(filter(lambda x: x.name == "Street 1", context.simulation.area.children))[0]
-        house1 = list(filter(lambda x: x.name == "S1 House 1", street1.children))[0]
-        permanent_load = list(filter(lambda x: x.name == "S1 H1 Load", house1.children))[0]
-        energy_profile = [ki for ki in permanent_load.strategy.state.desired_energy_Wh.values()]
-        assert all([permanent_load.strategy.energy == ei for ei in energy_profile])
+        assert "Street 1" in context.name_uuid_map.keys()
+        assert "S1 House 1" in context.name_uuid_map.keys()
+        assert "S1 H1 Load" in context.name_uuid_map.keys()
```

Comment:
```suggestion
        assert {"Street 1", "S1 House 1", "S1 H1 Load"}.is_subset(set(context.name_uuid_map.keys()))
```
I would also give this a try